# Assignment 6

To measure the psychological state of Americans, a psychology researcher (in 1929) devised
three questions put to 3443 Americans from different walks of life. He asked them whether they
were happy. They could answer only **yes** or **no** to each question. By today’s standards the study
is quite simple, and he did no formal statistical analysis of his data. The frequency data (number
of **yes** and **no** responses within each employment category) from this study are in the file
*Americandata.csv*.

---

You need to use the data and answer the question, that is, **Is the type of profession related to the happiness of American people?**
Report the **χ2** statistic and the *statistical significance of test*, in addition to the contingency table.
Also, explain the results and what they mean. What does the value of **χ2** statistic mean? Is it
good or bad? Is the test statistically significant?

In [9]:
import pandas as pd
from scipy import stats

df = pd.read_csv("./Americandata.csv")
df.head(20)

,Happy,Students,Laborers,Preachers,Physicians,Housewives,Teachers,Lawyers,Musicians
0,Yes,390,378,35,159,78,108,11,31
1,No,1610,122,265,51,122,38,64,19


In [10]:
contingency_table = df.set_index("Happy")

print(contingency_table)


chi2, p_value, dof, expected = stats.chi2_contingency(contingency_table, correction=False)

expected_df = pd.DataFrame(expected, index=contingency_table.index, columns=contingency_table.columns)

print("\nExpected Table:")
print(expected_df.round(2))

print(f"\nChi-square = {chi2:.2f}")
print(f"Degrees of freedom = {dof}")
print(f"p-value = {p_value:.2e}")

       Students  Laborers  Preachers  Physicians  Housewives  Teachers  \
Happy                                                                    
Yes         390       378         35         159          78       108   
No         1610       122        265          51         122        38   

       Lawyers  Musicians  
Happy                      
Yes         11         31  
No          64         19  

Expected Table:
       Students  Laborers  Preachers  Physicians  Housewives  Teachers  \
Happy                                                                    
Yes      683.71    170.93     102.56       71.79       68.37     49.91   
No      1316.29    329.07     197.44      138.21      131.63     96.09   

       Lawyers  Musicians  
Happy                      
Yes      25.64      17.09  
No       49.36      32.91  

Chi-square = 936.14
Degrees of freedom = 7
p-value = 7.52e-198


Put your analysis in the empty `markdown` cell below.

The chi-square statistic is 936.14 with a p-value of 7.52x10^-198. Since the P-value is less than 0.05, the test is statistically signifigant. This means theres a relationship between happiness and professions in the data. The large chi-square value shows that the observed results are very different from what we would expect if profession and happiness were unrelated.